src/pages/register/index.js

```diff
@@ -6,49 +6,80 @@ import CredentialsCard from "../../components/credentials";
 
 const Register = () => {
-	const { onRegister } = useAuth();
-	const [formData, setFormData] = useState({ email: "", password: "" });
+  const { onRegister } = useAuth();
+  const [formData, setFormData] = useState({ email: "", password: "" });
+  const [errorMsg, setErrorMsg] = useState("");
+  const specialChars = /^(?=.*[!@#$%^&*])/;
+  const numbers = /^(?=.*[0-9])/;
+  // This doesn't work and could be implemented later
+  // const forbiddenChars = /^(?=.*[(){}[]|`¬¦!"£$%^&*"<>:;#~_-+=,@])/;
+  const onChange = (e) => {
+    const { name, value } = e.target;
+    setFormData({ ...formData, [name]: value });
+  };
 
-	const onChange = (e) => {
-		const { name, value } = e.target;
-		setFormData({ ...formData, [name]: value });
-	};
-
-	return (
-		<div className="bg-blue register credentialpage">
-			<CredentialsCard
-				title="Register"
-				socialLinksTitle="Or sign up with"
-				altButtonTitle="Already a user?"
-				altButtonLink="/login"
-				altButtonText="Log in"
-			>
-				<div className="register-form">
-					<form>
-						<TextInput
-							value={formData.email}
-							onChange={onChange}
-                            type="email"
-							name="email"
-							label={"Email *"}
-						/>
-						<TextInput
-							value={formData.password}
-							onChange={onChange}
-							name="password"
-							label={"Password *"}
-							type={"password"}
-						/>
-					</form>
-					<Button
-						text="Sign up"
-						onClick={() => onRegister(formData.email, formData.password)}
-						classes="green width-full"
-					/>
-				</div>
-			</CredentialsCard>
-		</div>
-	);
+  return (
+    <div className="bg-blue register credentialpage">
+      <CredentialsCard
+        title="Register"
+        socialLinksTitle="Or sign up with"
+        altButtonTitle="Already a user?"
+        altButtonLink="/login"
+        altButtonText="Log in"
+      >
+        <div className="register-form">
+          <form>
+            <TextInput
+              value={formData.email}
+              onChange={onChange}
+              type="email"
+              name="email"
+              label={"Email *"}
+            />
+            <TextInput
+              value={formData.password}
+              onChange={onChange}
+              name="password"
+              label={"Password *"}
+              type={"password"}
+            />
+            <div>{errorMsg}</div>
+          </form>
+          <Button
+            text="Sign up"
+            onClick={() => {
+              let enteredPass = formData.password;
+              setErrorMsg("");
+              if (enteredPass.length < 8) {
+                setErrorMsg("Password must be at least 8 characters long!");
+              } else if (
+                enteredPass.toLowerCase() === enteredPass ||
+                enteredPass.toUpperCase() === enteredPass
+              ) {
+                setErrorMsg(
+                  "Password must contain at least 1 uppercase and 1 lowercase letter"
+                );
```

Comment:
This code is unreachable because of the _else if_, the condition is only checked if the password is less than 8.